### DISCUSSION TITLE
how do i fix Next.js Image for static images - "URL" PARAMETER IS VALID BUT UPSTREAM RESPONSE IS INVALID

### DISCUSSION THREAD

**Discussion by prateekshaweb** [vercel/next.js#62676] [Help]
### Summary

I am using next.js <Image> image component and uploading to a apache2 server.... I have install "sharp" ... but the image tag is giving a lot of errors and does not display... "URL" PARAMETER IS VALID BUT UPSTREAM RESPONSE IS INVALID

has anybody a fix for it?

**Comment by rlueder** [comment]
We fixed this by passing `loader={({ src }) => src}` to every instance of `Image` component. In our case the src is already being processed by another service.

```
<Image loader={({ src }) => src} {...otherProps} />
```

**Reply by bilofy** [reply]
Hey, this truly fixed the error but i wonder why actually?

**Comment by stella-sre** [comment]
This error typically occurs because **Next.js Image Optimization** doesn't work properly on traditional Apache servers. The built-in optimizer expects a Node.js environment to process images.

## Solutions

### Option 1: Use a custom loader

```jsx
<Image 
  loader={({ src }) => src} 
  src="/my-image.jpg"
  width={500}
  height={300}
  alt="Description"
/>
```

### Option 2: Disable optimization globally

In your `next.config.js`:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  images: {
    unoptimized: true,
  },
}

module.exports = nextConfig
```

### Option 3: Use an external image CDN

If you still need optimization, consider using a third-party service:

```js
// next.config.js
module.exports = {
  images: {
    loader: 'custom',
    loaderFile: './my-loader.js',
  },
}
```

```js
// my-loader.js
export default function myImageLoader({ src, width, quality }) {
  return `https://your-cdn.com/${src}?w=${width}&q=${quality || 75}`
}
```

Popular options include **Cloudinary**, **Imgix**, or **Cloudflare Images**.

---

### Why does this happen?

Next.js Image Optimization runs a serverless function that:
1. Fetches the original image
2. Resizes/optimizes it
3. Serves the optimized version

Apache doesn't support this out of the box since it requires Node.js runtime.

**Comment by InfoYaseer** [comment]
This error usually happens because Next.js tries to optimize the image,
but the image path is not being resolved correctly in a static setup.

If you are using a static image from the `public` folder, make sure:

1. The image is placed inside `/public`
   Example:
2. Use an absolute path starting with `/`
❌ Wrong:
```js
<Image src="images/logo.png" />
<Image src="/images/logo.png" width={300} height={200} alt="logo" />
